covid19_sfbayarea/data/marin_scraper.py

```diff
@@ -0,0 +1,303 @@
+	series = []
+
+	csv_strs = csv_.splitlines()
+	keys = csv_strs[0].split(',')
+	
+	if keys != ['Date', 'Total Cases', 'Total Recovered*', 'Total Hospitalized', 'Total Deaths']:
+		raise ValueError('The headers have changed')
+
+	case_history = []
+
+	for row in csv_strs[1:]:
+	 	daily = {}
+	 	# Grab the date in the first column
+	 	date_time_obj = datetime.strptime(row.split(',')[0], '%m/%d/%Y')
+	 	daily["date"] = date_time_obj.isoformat()
+	 	# Collect the case totals in order to compute the change in cases per day 
+	 	case_history.append(int(row.split(',')[1]))
+	 	# Grab the cumulative number in the fifth column
+	 	daily["cumul_cases"] = int(row.split(',')[1])
+	 	series.append(daily)
+		
+	case_history_diff = np.diff(case_history) 
+	# there will be no calculated difference for the first day, so adding it in manually
+	case_history_diff = np.insert(case_history_diff, 0, 0) 
+	# adding the case differences into the series
+	for val, case_num in enumerate(case_history_diff):
+		series[val]["cases"] = case_num
+	return series
+
+def get_death_series(chart_id: str, url: str) -> List:
+	"""This method extracts the date, number of cumulative deaths, and new deaths."""
+	csv_ = extract_csvs(chart_id, url)
+	series = []
+
+	csv_strs = csv_.splitlines()
+	keys = csv_strs[0].split(',')
+	if keys != ['Date', 'Total Cases', 'Total Recovered*', 'Total Hospitalized', 'Total Deaths']:
+		raise ValueError('The headers have changed.')
+
+	death_history = []
+
+	for row in csv_strs[1:]:
+	 	daily = {}
+	 	# Grab the date in the first column
+	 	date_time_obj = datetime.strptime(row.split(',')[0], '%m/%d/%Y')
+	 	daily["date"] = date_time_obj.isoformat()
+	 	# Collect the death totals in order to compute the change in deaths per day 
+	 	death_history.append(int(row.split(',')[4]))
+	 	# Grab the cumulative number in the fifth column
+	 	daily["cumul_deaths"] = int(row.split(',')[4])
+	 	series.append(daily)
+		
+	death_history_diff = np.diff(death_history) 
+	# there will be no calculated difference for the first day, so adding it in manually
+	death_history_diff = np.insert(death_history_diff, 0, 0) 
+	# adding the case differences into the series
+	for val, death_num in enumerate(death_history_diff):
+		series[val]["deaths"] = death_num
+	return series
+
+def get_breakdown_age(chart_id: str, url: str) -> Tuple:
+	"""This method gets the breakdown of cases and deaths by age."""
+	csv_ = extract_csvs(chart_id, url)
+	c_brkdown = []
+	d_brkdown = []
+
+	csv_strs = csv_.splitlines()
+	keys = csv_strs[0].split(',') 
+
+	if keys != ['Age Category', 'POPULATION', 'Cases', 'Hospitalizations', 'Deaths']:
+		raise ValueError('The headers have changed')
+
+	ages = ['0-18', '19-34', '35-49', '50-64', '65+'] 
+	for row in csv_strs[1:]:
+		c_age = {}
+		d_age = {}
+		# Extracting the age group and the raw count (the 3rd and 5th columns, respectively) for both cases and deaths.
+		# Each new row has data for a different age group.
+		c_age["group"] = row.split(',')[0]
```

Comment:
These still need to get mapped to the group names from https://github.com/sfbrigade/data-covid19-sfbayarea/tree/master/data_models, right?

Reply:
yes, thanks for catching this Rob!!